lib/Analysis/Utility.cpp

```diff
@@ -714,8 +714,7 @@ bool matchMmaV3AndDotOperandLayout(RankedTensorType srcTy,
   auto ans = mmaLayout.getVersionMajor() == 3 &&
              dotOperandLayout.getOpIdx() == 0 &&
              mmaLayout.getWarpsPerCTA()[1] == 1 &&
-             !cvtNeedsSharedMemory(parentTy, srcTy) &&
-             (elementTypeSize == 16 || elementTypeSize == 8) &&
+             !cvtNeedsSharedMemory(parentTy, srcTy) && elementTypeSize == 8 &&
```

Comment:
The `elementTypeSize == 16` path was actually buggy in the legacy code. Instead of marking it as a no-op, which is the behavior of the legacy conversion, some shuffle operations between registers are supposed to happen. Making it use the current code path can perfectly solve the problem